tests/initializer/phare_dict.cpp

```diff
@@ -0,0 +1,112 @@
+
+
+    dict["ions"]["pop0"]["particle_initializer"]["nbrPartPerCell"] = int{100};
+    dict["ions"]["pop0"]["particle_initializer"]["charge"]         = -1.;
+    dict["ions"]["pop0"]["particle_initializer"]["basis"]          = std::string{"Cartesian"};
+
+    dict["ions"]["pop1"]["name"] = std::string{"alpha"};
+    dict["ions"]["pop1"]["mass"] = 1.;
+    dict["ions"]["pop1"]["particle_initializer"]["name"]
+        = std::string{"MaxwellianParticleInitializer"};
+    dict["ions"]["pop1"]["particle_initializer"]["density"] = static_cast<InitFunctionT>(density);
+
+    dict["ions"]["pop1"]["particle_initializer"]["bulk_velocity_x"]
+        = static_cast<InitFunctionT>(vx);
+
+    dict["ions"]["pop1"]["particle_initializer"]["bulk_velocity_y"]
+        = static_cast<InitFunctionT>(vy);
+
+    dict["ions"]["pop1"]["particle_initializer"]["bulk_velocity_z"]
+        = static_cast<InitFunctionT>(vz);
+
+
+    dict["ions"]["pop1"]["particle_initializer"]["thermal_velocity_x"]
+        = static_cast<InitFunctionT>(vthx);
+
+    dict["ions"]["pop1"]["particle_initializer"]["thermal_velocity_y"]
+        = static_cast<InitFunctionT>(vthy);
+
+    dict["ions"]["pop1"]["particle_initializer"]["thermal_velocity_z"]
+        = static_cast<InitFunctionT>(vthz);
+
+
+    dict["ions"]["pop1"]["particle_initializer"]["nbrPartPerCell"] = int{100};
+    dict["ions"]["pop1"]["particle_initializer"]["charge"]         = -1.;
+    dict["ions"]["pop1"]["particle_initializer"]["basis"]          = std::string{"Cartesian"};
+
+    dict["electromag"]["name"]             = std::string{"EM"};
+    dict["electromag"]["electric"]["name"] = std::string{"E"};
+    dict["electromag"]["magnetic"]["name"] = std::string{"B"};
+
+    dict["electromag"]["magnetic"]["initializer"]["x_component"] = static_cast<InitFunctionT>(bx);
+    dict["electromag"]["magnetic"]["initializer"]["y_component"] = static_cast<InitFunctionT>(by);
+    dict["electromag"]["magnetic"]["initializer"]["z_component"] = static_cast<InitFunctionT>(bz);
+
+    dict["electrons"]["pressure_closure"]["name"] = std::string{"isothermal"};
+    dict["electrons"]["pressure_closure"]["Te"]   = 0.12;
+
+    return dict;
+}
+
+
+
+int main(int argc, char** argv)
+{
+    std::string const filename = "pharedict.dat";
+    PHARE::initializer::dump_phare_dict(createDict(), filename);
+    // auto const dict = PHARE::initializer::load_phare_dict(filename);
+
+    auto const dict = PHARE::initializer::PHAREDictHandler::load(filename);
+
+    int r = 0;
+
+    r += dict["ions"]["nbrPopulations"].to<std::size_t>() != 2;
+
+    r += dict["ions"]["pop0"]["particle_initializer"]["nbrPartPerCell"].to<int>() != 100;
+    r += dict["ions"]["pop1"]["particle_initializer"]["nbrPartPerCell"].to<int>() != 100;
+
+    r += dict["electromag"]["magnetic"]["initializer"]["x_component"].isValue(); // NOT SERIALIZABLE
+
+    r += dict["electrons"]["pressure_closure"]["Te"].to<double>() != 0.12;
+
+    return r;
```

Comment:
is a non-zero return value visible as an error in our tests?
I don't see this test added in the cmake tests, should probably be included (as a gtest)?